core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/workspace/runners/CloneWorkspaceContainerRunnerSpec.scala

```diff
@@ -0,0 +1,219 @@
+  val wsCreatedDate: DateTime = DateTime.parse("2023-01-18T10:08:48.541-05:00")
+  val monitorRecord: WorkspaceManagerResourceMonitorRecord =
+    WorkspaceManagerResourceMonitorRecord.forCloneWorkspaceContainer(
+      UUID.randomUUID(),
+      workspaceId,
+      RawlsUserEmail(userEmail)
+    )
+
+  val workspace: Workspace = Workspace(
+    "test-ws-namespace",
+    "test-ws-name",
+    workspaceId.toString,
+    "test-bucket",
+    None,
+    wsCreatedDate,
+    wsCreatedDate,
+    "a_user",
+    Map()
+  )
+}
+
+class CloneWorkspaceContainerRunnerSpec extends AnyFlatSpecLike with MockitoSugar with Matchers with ScalaFutures {
+  implicit val executionContext: ExecutionContext = TestExecutionContext.testExecutionContext
+
+  behavior of "initial setup and basic requirements updating workspace container cloning status monitoring"
+
+  it should "return a completed status if the workspace id is not set on the job" in {
+    val runner = new CloneWorkspaceContainerRunner(
+      mock[SamDAO],
+      mock[WorkspaceManagerDAO],
+      mock[SlickDataSource],
+      mock[GoogleServicesDAO]
+    )
+    whenReady(runner(monitorRecord.copy(workspaceId = None)))(
+      _ shouldBe WorkspaceManagerResourceMonitorRecord.Complete
+    )
+  }
+
+  it should "return a completed status if no workspace is found for the id" in {
+    val runner = spy(
+      new CloneWorkspaceContainerRunner(
+        mock[SamDAO],
+        mock[WorkspaceManagerDAO],
+        mock[SlickDataSource],
+        mock[GoogleServicesDAO]
+      )
+    )
+    val completedTime = "2023-01-16T10:08:48.541-05:00"
+    val expectedTime = DateTime.parse(completedTime)
+    val report = new JobReport().status(JobReport.StatusEnum.SUCCEEDED).completed(completedTime)
+
+    doAnswer { answer =>
+      val specifiedTime = answer.getArgument(1).asInstanceOf[DateTime]
+      specifiedTime shouldBe expectedTime
+      Future.successful(None)
+    }.when(runner)
+      .cloneSuccess(ArgumentMatchers.eq(workspaceId), ArgumentMatchers.eq(expectedTime))(
+        ArgumentMatchers.any[ExecutionContext]()
+      )
+
+    whenReady(runner.handleCloneResult(workspaceId, report))(
+      _ shouldBe WorkspaceManagerResourceMonitorRecord.Complete
+    )
+
+    verify(runner)
+      .cloneSuccess(ArgumentMatchers.eq(workspaceId), ArgumentMatchers.eq(expectedTime))(
+        ArgumentMatchers.any[ExecutionContext]()
+      )
+  }
+
+  it should "return a completed status if no user email is set on the job" in {
+    val runner = spy(
+      new CloneWorkspaceContainerRunner(
+        mock[SamDAO],
+        mock[WorkspaceManagerDAO],
+        mock[SlickDataSource],
+        mock[GoogleServicesDAO]
+      )
+    )
+
+    doAnswer(answer =>
+      Future.successful(Some(workspace.copy(errorMessage = Some(answer.getArgument(1).asInstanceOf[String]))))
+    ).when(runner)
+      .cloneFail(ArgumentMatchers.any(), ArgumentMatchers.any())(ArgumentMatchers.any[ExecutionContext]())
```

Comment:
Will this test fail if `cloneFail` isn't called?

Reply:
I added a verify check, so it will now